concent_api/core/tests/test_core_views.py

```diff
@@ -286,6 +286,93 @@ def test_send_reject_message_save_as_receive_out_of_band_status(self):
         self.assertEqual(reject_response.status_code, 202)
         self.assertEqual(Message.objects.last().type, message.RejectReportComputedTask.TYPE)
 
+    def test_send_should_return_http_400_if_task_to_compute_deadline_is_not_an_integer(self):
+        compute_task_def = message.ComputeTaskDef()
+        compute_task_def['task_id'] = '8'
+
+        invalid_values = [
+            -11,
+            'a11b',
+            {},
+            [],
+            (1, 2, 3),
+            None,
+            False,
+        ]
+
+        for deadline in invalid_values:
+            Message.objects.all().delete()
+            task_to_compute = message.TaskToCompute(
+                timestamp = int(datetime.datetime.now().timestamp()),
+                compute_task_def = self.compute_task_def,
+            )
+
+            serialized_task_to_compute      = dump(task_to_compute,             REQUESTOR_PRIVATE_KEY,   PROVIDER_PUBLIC_KEY)
+            deserialized_task_to_compute    = load(serialized_task_to_compute,  PROVIDER_PRIVATE_KEY,  REQUESTOR_PUBLIC_KEY, check_time = False)
+
+            deserialized_task_to_compute.compute_task_def['deadline'] = deadline
```

Comment:
Why change this after serialization/deserialization and not in the original `task_to_compute`? You now have `deserialized_task_to_compute` that's different than the original `task_to_compute`. Fortunately you're not using it for anything in this test but if you were modify the test later, this can only lead to problems.

Reply:
The issue for that is what I have described
> Btw. something must have changed in golem messages, because now giving negative timestamp changes it into now() during serialization.

So when I set it before serializing, it gets changed and that would not really test the case. So I can either move it up front and assume the correctness because of serialization, or keep it that way until golem team will resolve bug.

Reply:
Oh, OK. But add at least a FIXME and a comment explaining this.

Reply:
Ok, there is no bug, it was just my mistake to use self.compute_task_def.